.github/workflows/go.yml

```diff
@@ -44,7 +44,42 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
       - name: Run tests with coverage
-        run: go test -v -race -coverprofile=coverage -covermode=atomic ./...
+        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run !TestGobSerialization
+      - name: Upload coverage report to Codecov
+        uses: codecov/codecov-action@v1.5.0
+        with:
+          file: ./coverage
+          flags: unittests
+
+  redis:
+    name: Redis
+    strategy:
+      matrix:
+        go-version: [ 1.18.x, 1.19.x ]
+        platform: [ ubuntu-latest ]
+    runs-on: ${{ matrix.platform }}
+    services:
+      redis:
+        image: redis:4
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run tests with coverage
+        run: go test -v -race -coverprofile=coverage -covermode=atomic ./... -run ^\QTestGobSerialization\E$
```

Comment:
I don't think this regep is working? 

```
testing: warning: no tests to run
```

https://github.com/flamego/csrf/actions/runs/3533397520/jobs/5928964766#step:5:25

Reply:
Yes, I've also noticed something wrong with the regex.
This PR is still a work in progress. 😥

Reply:
I fixed the test run, but I don't think it makes to _not_ run the full test suite unless it becomes painfully long.